chia/wallet/puzzles/dao_proposal_timer.clsp

```diff
@@ -40,6 +40,11 @@
   ; proposal_yes_votes
   ; proposal_total_votes
   ; proposal_innerpuzhash
+  ; [REVIEW - optimization] We're passing in a _lot_ of stuff here to simply validate the proposal timelock is correct
+  ; [^] Could we either a) curry in the timelock to this puzzle
+  ; [^] b) have the proposal announce the timelock to us and verify it that way?
```

Comment:
this is what's happening

Reply:
Er, perhaps I meant that we can just get away with currying in the proposal's singleton information since none of the inner puzzle stuff matters to us, so you can just pass in the whole hash of that.